### ISSUE TITLE
File format change due to __VERSION__ change

### ISSUE THREAD

**Issue by madwax** [ndevenish/Blender_ioEDM#8]
I'm doing a C++ import library to get a better understanding of the file format, I'm not a Python coder and TBH I hate the language.  Anyway  model::rootnode->property["__VERSION__"] can now be 2 or 3.
if( __VERSION__ == 3 ) then unknownA is NOT present so just start to read the bounding boxes.

**Comment by DartShinigami** [comment]
> I'm doing a C++ import library to get a better understanding of the file format, I'm not a Python coder and TBH I hate the language. Anyway model::rootnode->property["**VERSION**"] can now be 2 or 3.
> if( **VERSION** == 3 ) then unknownA is NOT present so just start to read the bounding boxes.

Are you cointributing to the code??? were I could find you build.

Thanks in advance

**Comment by madwax** [comment]
Once I get what .edm files I have passing my C++ code I will push it into repo.  As for the python import code not yet.  It should not be to hard to find the bit of code that imports the model::RootNode.  I just don't have the time or ability to check any python code as I only have blender 2.8 and the plugin is not even showing up (think the import/export API changed) Soz

**Comment by ndevenish** [comment]
RootNode is read here:
https://github.com/ndevenish/Blender_ioEDM/blob/79ebb79cec79d5b6be65411899c6301b9eef7452/io_EDM/edm/types.py#L316-L327
It looks like I didn't validate that the rootnode version stayed the same.

I'm not surprised that blender might not be reading the plugin, it's been quite a while since I had time to work on this/play DCS (I got a job shortly after).

**Comment by DartShinigami** [comment]
I tried the new code and worked with 2.8 thanks!!!!!!! BTW any news on updating more of the code to a new commit????

Again thanks

**Comment by madwax** [comment]
@DartShinigami  How have you gotten the plugin to work with 2.8? From what I'v read there are major  breaking changes all over the place. (Like light now called Lamp)

@ndevenish Thanks for getting this far :) Real life can always be a kicker on personal projects

**Comment by DartShinigami** [comment]
Well to be clear I have 2.7 with the plugin working with its limitations then I updated to 2.8 and is still working, also I downloaded the last one with the code change and override the plugin directory and it's working so far with the normal glishes the plugin has.

**Comment by madwax** [comment]
Interesting I'm a clean 2.8 and its not working for me. + I'm on a mac

**Comment by DartShinigami** [comment]
> Interesting I'm a clean 2.8 and its not working for me. + I'm on a mac

Windows here, and the version I had before update was 2.79. 

Regards

BTW please keep improving the plugin is very neat.

**Comment by damo740** [comment]
Hi All,
This is broken for me as well. I have tried fresh install into 2.79 but still does not work
@DartShinigami what do you mean "code change and override the plugin directory". What code change are you speaking of and how do you override the plugin directory?

@ndevenish Thanks for a great plugin.

**Comment by DartShinigami** [comment]
I just locate the plugin folder in the blender directory and copy paste and override the folders.
On later test I found that the plugin in blender 2.8 is not working properly as I said in previus post so I roll back to 2.79.

BTW the code change isn't mine is @ndevendish that changed the pluging with the code correction of @madwax

**Comment by damo740** [comment]
Thanks for your reply @DartShinigami. I follow your instructions but still
get the attached error message when trying to load models.

![BlenderIssue](https://user-images.githubusercontent.com/28345461/58215996-d0bd7680-7d3f-11e9-9393-fc180ad69eef.jpg)

**Comment by DartShinigami** [comment]
If you are trying to open a model like the new F-18 it would fail, cause the pluging doesn't open the new model version we have to wait for @ndevenish maybe he helps us with that.

Best Regards

**Comment by madwax** [comment]
I have been working on a 2.8 version of the plugin while on holiday (https://github.com/madwax/Blender_ioEDM branch:road_to_28).  I've not been able to test it with all the models yet as I've only got back but it was loading the Harrier and A10C cockpits so should load the F-18.    I'm still in the middle of trying to work out position issues (Load the harrier cockpit) and the bone/Skin stuff.  Also not touched the export side of things.

**Comment by madwax** [comment]
@damo740 Just looked at your error, my branch should work as I've fixed the bug your running into which is a 2.7->2.8 API change thing.

**Comment by ndevenish** [comment]
The newer models probably use features that weren't available when I wrote this, so not surprised they might have troubles

**Comment by DartShinigami** [comment]
> @damo740 Just looked at your error, my branch should work as I've fixed the bug your running into which is a 2.7->2.8 API change thing.

I'll try it today, and get back to you guys thanks again!!!!!

**Comment by damo740** [comment]
Hi Guys,

Thanks heaps for all your help. Thought I should let you know I am using blender 2.79b. I was able, when it first came out, open the new F-18C. It was a couple of months later when I tried to open the .edm file again that it wouldn't work.

@madwax How do I download your version of the code? I am new to GitHub and not sure how it all works.

Thanks again, guys.

Cheers
Damo

**Comment by madwax** [comment]
@damo740 I've been deving using the read.py script as it's just simpler for me.  The best thing to do it install the existing copy, download the source from my branch and overwrite all the files.  However I'm very new to blender & don't know much about it.  Also if your not using 2.8 don't use my branch.  The bug I think you have I through was a 2.79 -> 2.80 problem but it might well not have been.

**Comment by DartShinigami** [comment]
@madwax I tried your last version BTW sorry for late response, Here is the printscreen of the issue.

When you try to load the file, it only works partialy 

Best Regards 

Dunno why I doesn't let me upload the image I have to upload it to some place else.

https://ibb.co/HBgN0QD

**Comment by nuclearPigeon123** [comment]
Hey, just so you guys know, meshes created with the EDM exporter tool created by @ndevenish no longer work in the latest release of DCS. Something to do with the version only supporting meshes created with a certain version (or newer) of the plugin for 3dsMax. I can verfiy that meshes I created prior to the previous version with I have used in DCS, no longer work with DCS 2.5.5. I hope this info helps anyone working on this plugin.

**Comment by DartShinigami** [comment]
Hi guys:

I don't know if it hack or just a bug but is working for me, BTW **working with road to 28, bone support** the models that give error when loading like for example merkava or F-18 just try a second time to load after first time I will give an error but i'll load the model with the normal problems that @madwax and @ndevenish are working out.  

Hope it helps

I'll try to load the screenshot to other place cause github doesn't allow me to... dunno why.

**Comment by madwax** [comment]
@DartShinigami Sorry for not getting back to you sooner over the F-18 model.  Don't know why that's happening and without the file I can't say much more. tbh my time has been pulled into other things (like DCS lol) however with the current patch breaking so much I'll have to sometime.

@nuclearPigeon123 That's interesting.

**Comment by nuclearPigeon123** [comment]
Interesting indeed. CHeck the ED forums. Also if you go to where the official plugins are, the list is significantly smaller.
ftp://srv0files.eagle.ru/mods/edm_plugins/

I should also mention this has also effected mods created using 3dsMax using the older export tools in the same way.

Sadly for me none of my meshes work in 2.5.5. I was doing a video series on my attempt to make a mod using free tools, but I have had to conclude it for now. Are any contributors here planning on restoring the functionality of this amazing tool? I would love to contribute personally, but I would not know where to start, as I have only ever done .net c# for web.

**Comment by madwax** [comment]
@nuclearPigeon123 I'm creating a new issue for the exporter.  I started looking at it last night but I've had to switch from 2.8 to 2.7. By the way are you doing the BD-5J on YouTube?

**Comment by nuclearPigeon123** [comment]
Yes, that is me. I will be announcing hopefully this evening, that I will be continuing with the project, but switching to XPlane 11, as Xplane has an official plugin Xplane2Blender.  This is also only supporting Blender 2.78.

If you gentlemen get this working and exporting with compatibility for DCS 2.5.5, I will hopefully have a model that I can rig for DCS also.

I am watching this with keen interest, and wish you the best of luck.

**Comment by madwax** [comment]
@nuclearPigeon123 I'm motivated to get it exporting mainly only just started looking to DCS modding because I want to do the Lightning, I can't model but trying to get my hands on any aeor data I can.

**Comment by DartShinigami** [comment]
> @nuclearPigeon123 I'm motivated to get it exporting mainly only just started looking to DCS modding because I want to do the Lightning, I can't model but trying to get my hands on any aeor data I can.

But please don't forgett the import cause it helps at least in my point of view how other people do something to learn how to do it your selft, also there are some other dudes that like moding and this import/export is a must for them.

best regards

**Comment by nuclearPigeon123** [comment]
@madwax https://youtu.be/62fhY4WFK2s?t=60, I have an EE Lightning F6. I have a very basic cockpit with 1 gauge. I have basic LUA in place to run the model in DCS. I don't have a programmer ;)

**Comment by nuclearPigeon123** [comment]
@madwax, any joy on this issue? I would be very happy to work with you on the lightning, if you are in a position to bring this tool upto date.

**Comment by madwax** [comment]
@nuclearPigeon123 Soz not getting back to you sooner. I'm knee deap in the write code currently but I'm fighting issues like actions crashing the viewer and transforms are all over the place. no eta tbo :(  As for the Lightning hell yes (my email is public so get in touch), I've got a lead on flight manuals and such info.  That being said I want export working first

**Comment by Hawkeye-CH** [comment]
Hi

@DartShinigami 
Could you contact me on reddit.com (Hawkeye_CH, with underscore)

I saw what you wrote and would be glad if you can help me maybe with it:
>  don't know if it hack or just a bug but is working for me, BTW working with road to 28, bone support the models that give error when loading like for example merkava or F-18 just try a second time to load after first time I will give an error but i'll load the model with the normal problems that @madwax and @ndevenish are working out.
